pageserver/src/walreceiver/walreceiver_connection.rs

```diff
@@ -29,12 +30,29 @@ use crate::{
 use postgres_ffi::waldecoder::WalStreamDecoder;
 use utils::{lsn::Lsn, pq_proto::ReplicationFeedback, zid::ZTenantTimelineId};
 
+/// Status of the connection.
+#[derive(Debug, Clone)]
+pub struct WalConnectionStatus {
+    /// If we were able to initiate a postgres connection, this means that safekeeper process is at least running.
+    pub is_connected: bool,
+    /// Defines a healthy connection as one on which we have received at least some WAL bytes.
+    pub is_received_wal: bool,
```

Comment:
Maybe `wal_received`? Or `has_received_wal`?

Alternatively, we could name everything as you have in the comments:
* `connection_healthy: bool`
* `safekeeper_lsn: Option<Lsn>`
* etc.